llama_index/indices/struct_store/json_query.py

```diff
@@ -0,0 +1,186 @@
+import logging
+from typing import Any, Union, Optional, Dict, Callable, List
+import json
+from langchain.input import print_text
+
+from llama_index.indices.query.base import BaseQueryEngine
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.indices.service_context import ServiceContext
+from llama_index.prompts.default_prompts import DEFAULT_JSON_PATH_PROMPT
+from llama_index.token_counter.token_counter import llm_token_counter
+from llama_index.response.schema import Response
+from llama_index.prompts.base import Prompt
+from llama_index.prompts.prompt_type import PromptType
+
+logger = logging.getLogger(__name__)
+IMPORT_ERROR_MSG = (
+    "`jsonpath_ng` package not found, please run `pip install jsonpath-ng`"
+)
+
+JSONType = Union[Dict[str, "JSONType"], List["JSONType"], str, int, float, bool, None]
+
+
+DEFAULT_RESPONSE_SYNTHESIS_PROMPT_TMPL = (
```

Comment:
very nit: we don't have a consistent policy for where to put prompts right now. Some are in the prompts files, some are inline in the query engine definition.